### PR TITLE
Add imports for extension add methods

### LINKED ISSUE #269: Collection Initializers don't show Quick Fixes to import extension methods
Ported from TFS WorkItem: <b>1105748</b>

------------------------------

@SSL reported:</i><br><br>Source:<br><br>
```C#
class X : IEnumerable 
{
       public IEnumerator GetEnumerator() {		
          new X { 1 };
          return null;
	}
}
namespace Ext {
	static class Extensions {
		public static void Add(this X x, int i) { }
	}
}
```
The `new X` callsite should show a quick fix for `using Ext`.
-----------------------------

### PR DESCRIPTION
**Pull request by jmarolf**
Offer to add imports\namespaces for extension methods that apply to collection initializers.

Fixes #269

### CODE REVIEW

**Comment by jmarolf** [comment]
@srivatsn @tmeschter @shyamnamboodiripad @mavasani @heejaechang @JohnHamby Could you take a look, please?

**Comment by jmarolf** [comment]
@dotnet-bot test this please.

**Comment by jmarolf** [comment]
@dotnet-bot test this please.

File: src/Features/Core/CodeFixes/AddImport/AbstractAddImportCodeFixProvider.cs [hunk 1]
```diff
@@ -305,16 +317,51 @@ private IEnumerable<INamespaceSymbol> FilterForFieldsAndProperties(SemanticModel
             // See if the name binds.  If it does, there's nothing further we need to do.
             if (ExpressionBinds(node, semanticModel, cancellationToken, checkForExtensionMethods: true))
             {
-                return null;
+                return Task.FromResult(SpecializedCollections.EmptyEnumerable<ISymbol>());
```
Comment (heejaechang) [inline src/Features/Core/CodeFixes/AddImport/AbstractAddImportCodeFixProvider.cs]: use SpecializedTasks

File: src/Features/Core/CodeFixes/AddImport/AbstractAddImportCodeFixProvider.cs [hunk 2]
```diff
@@ -305,16 +317,51 @@ private IEnumerable<INamespaceSymbol> FilterForFieldsAndProperties(SemanticModel
...
             }
 
             string name;
             int arity;
             syntaxFacts.GetNameAndArityOfSimpleName(node, out name, out arity);
+            if (name == null)
+            {
+                return Task.FromResult(SpecializedCollections.EmptyEnumerable<ISymbol>());
```
Comment (heejaechang) [inline src/Features/Core/CodeFixes/AddImport/AbstractAddImportCodeFixProvider.cs]: same here

**Comment by tmeschter** [comment]
:+1:

**Comment by heejaechang** [comment]
:+1

**Comment by heejaechang** [comment]
sorry for late review.

**Pull request merged**